src/Compilers/CSharp/Portable/Symbols/PublicModel/Symbol.cs

```diff
@@ -0,0 +1,245 @@
+        {
+            return this.Equals(other as Symbol, equalityComparer);
+        }
+
+        protected bool Equals(Symbol other, CodeAnalysis.SymbolEqualityComparer equalityComparer)
+        {
+            return other is object && UnderlyingSymbol.Equals(other.UnderlyingSymbol, equalityComparer.CompareKind);
+        }
+
+        ImmutableArray<Location> ISymbol.Locations
+        {
+            get
+            {
+                return UnderlyingSymbol.Locations;
+            }
+        }
+
+        ImmutableArray<SyntaxReference> ISymbol.DeclaringSyntaxReferences
+        {
+            get
+            {
+                return UnderlyingSymbol.DeclaringSyntaxReferences;
+            }
+        }
+
+        ImmutableArray<AttributeData> ISymbol.GetAttributes()
+        {
+            return StaticCast<AttributeData>.From(UnderlyingSymbol.GetAttributes());
+        }
+
+        Accessibility ISymbol.DeclaredAccessibility
+        {
+            get
+            {
+                return UnderlyingSymbol.DeclaredAccessibility;
+            }
+        }
+
+        void ISymbol.Accept(SymbolVisitor visitor)
+        {
+            Accept(visitor);
+        }
+
+        protected abstract void Accept(SymbolVisitor visitor);
+
+        TResult ISymbol.Accept<TResult>(SymbolVisitor<TResult> visitor)
+        {
+            return Accept(visitor);
+        }
+
+        protected abstract TResult Accept<TResult>(SymbolVisitor<TResult> visitor);
+
+        string ISymbol.GetDocumentationCommentId()
+        {
+            return UnderlyingSymbol.GetDocumentationCommentId();
+        }
+
+        string ISymbol.GetDocumentationCommentXml(CultureInfo preferredCulture, bool expandIncludes, CancellationToken cancellationToken)
+        {
+            return UnderlyingSymbol.GetDocumentationCommentXml(preferredCulture, expandIncludes, cancellationToken);
+        }
+
+        string ISymbol.ToDisplayString(SymbolDisplayFormat format)
+        {
+            return SymbolDisplay.ToDisplayString(this, format);
+        }
+
+        ImmutableArray<SymbolDisplayPart> ISymbol.ToDisplayParts(SymbolDisplayFormat format)
+        {
+            return SymbolDisplay.ToDisplayParts(this, format);
+        }
+
+        string ISymbol.ToMinimalDisplayString(SemanticModel semanticModel, int position, SymbolDisplayFormat format)
+        {
+            return SymbolDisplay.ToMinimalDisplayString(this, GetCSharpSemanticModel(semanticModel), position, format);
+        }
+
+        ImmutableArray<SymbolDisplayPart> ISymbol.ToMinimalDisplayParts(SemanticModel semanticModel, int position, SymbolDisplayFormat format)
+        {
+            return SymbolDisplay.ToMinimalDisplayParts(this, GetCSharpSemanticModel(semanticModel), position, format);
+        }
+
+        internal static CSharpSemanticModel GetCSharpSemanticModel(SemanticModel semanticModel)
+        {
+            var csharpModel = semanticModel as CSharpSemanticModel;
```

Comment:
Nit: could just do `if (semanticModel is CSharpSemanticModel csharpModel)`

Reply:
> Nit: could just do if (semanticModel is CSharpSemanticModel csharpModel)

This code is moved from somewhere else.

---
In reply to: [338819338](https://github.com/dotnet/roslyn/pull/39498#discussion_r338819338) [](ancestors = 338819338)